# 3. Inspect ADM2 boundary geometry

This notebook checks whether the downloaded ADM2 boundaries look usable before any raster overlay or mapping.


## Load boundaries


In [5]:
from pathlib import Path
import sys

NOTEBOOK_DIR = Path.cwd()
PROJECT_ROOT = NOTEBOOK_DIR.parent if NOTEBOOK_DIR.name == "notebooks" else NOTEBOOK_DIR
SRC_DIR = PROJECT_ROOT / "src"
if str(SRC_DIR) not in sys.path:
    sys.path.append(str(SRC_DIR))

import geopandas as gpd

from childreach.paths import ADM2_BOUNDARIES_GEOJSON

boundaries = gpd.read_file(ADM2_BOUNDARIES_GEOJSON)

print("Rows:", len(boundaries))
print("CRS:", boundaries.crs)
print(boundaries[["shapeName", "shapeID", "shapeType", "shapeGroup"]].head())


Rows: 159
CRS: EPSG:4326
      shapeName                  shapeID shapeType shapeGroup
0  Alto Molocue  85939544B55280380491119      ADM2        MOZ
1       Ancuabe  85939544B12924931865401      ADM2        MOZ
2       Angoche  85939544B28799715012212      ADM2        MOZ
3       Angonia  85939544B12573256040215      ADM2        MOZ
4        Balama  85939544B22682737370097      ADM2        MOZ


## Validate key geometry and identifier fields

These checks help catch broken geometries, duplicate IDs, missing names, or an unexpected administrative level.


In [6]:
print("Missing geometries:", boundaries.geometry.isna().sum())
print("Invalid geometries:", (~boundaries.geometry.is_valid).sum())

print("Unique shapeIDs:", boundaries["shapeID"].nunique())
print("Duplicate shapeIDs:", boundaries["shapeID"].duplicated().sum())

print("Missing shapeNames:", boundaries["shapeName"].isna().sum())
print("Unique shapeNames:", boundaries["shapeName"].nunique())

print("shapeType values:")
print(boundaries["shapeType"].value_counts())

print("shapeGroup values:")
print(boundaries["shapeGroup"].value_counts())


Missing geometries: 0
Invalid geometries: 0
Unique shapeIDs: 159
Duplicate shapeIDs: 0
Missing shapeNames: 0
Unique shapeNames: 159
shapeType values:
shapeType
ADM2    159
Name: count, dtype: int64
shapeGroup values:
shapeGroup
MOZ    159
Name: count, dtype: int64


## Interpretation

The boundary file should contain 159 valid ADM2 features in EPSG:4326, with unique `shapeID` values. These identifiers become the join key for district-level outputs.
